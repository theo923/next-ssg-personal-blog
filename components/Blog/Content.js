import React from 'react';
import AuthorInfo from './AuthorSideBar';
import Body from './Body';
import CategoriesSideBar from './CategoriesSideBar';
import ProjectInfo from '../Project/ProjectInfo';


const Content = ({type, category, data, item, httpString, categories}) => {
    return (
        <div className='h-full'>
            {type === 'blog' ?
                <div>
                    <div id='header' className='flex items-center justify-center shadow-lg bg-gradient-to-r from-blue-500 to-blue-600' style={{height: 400}}></div>
                    <div className='grid grid-cols-1 lg:grid-cols-3 2xl:grid-cols-5 '>
                        <div className='order-2 2xl:order-1 lg:col-span-3 2xl:col-span-1 block justify-self-center'><CategoriesSideBar categories={categories} /></div>
                        <div className='2xl:order-2 lg:col-span-2 2xl:col-span-3'><Body category={category} data={data} item={item} httpString={httpString} /></div>
                        <div className='2xl:order-3 justify-self-center '><AuthorInfo /></div>
                    </div>
                </div> : 
            type === 'project' ? 
                <div id='header' className='flex items-center justify-center shadow-lg bg-gradient-to-r from-purple-400 to-purple-500'>
                    <ProjectInfo />
                </div> 
            : null}
        </div>

    )
    
}

export default Content;
