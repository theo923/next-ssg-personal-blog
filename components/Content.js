import React from 'react';


const Content = ({item}) => {
    return (
        <div className='h-screen' >
            <div id='header' className='flex items-center justify-center bg-gradient-to-tr from-blue-100 via-blue-300 to-blue-500' style={{height: 300}}>
                <div className=''>Image with header</div>
            </div>
            <div className='grid grid-cols-3 justify justify-items-center'>
                <div className=''>5345435</div>
                <div className=''>{item}</div>
                <div className=''>5345435</div>
                
            </div>
        </div>

    )
    
}

export default Content;
