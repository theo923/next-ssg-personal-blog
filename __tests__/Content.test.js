import { shallow } from 'enzyme';
import React from 'react';
import Content from '../components/Content';

describe('Content test', () => {

    it("expect Content component is rendered without crashing", () => {
        const wrapper = shallow(<Content />);   
    });

    it("expect Content component is rendered && match with the Snapshot", () => {
        const wrapper = shallow(<Content />);   
        expect(shallow(<Content />)).toMatchSnapshot();
    });
    
});




