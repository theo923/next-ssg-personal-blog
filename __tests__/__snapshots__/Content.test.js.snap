// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content test expect Content component is rendered && match with the Snapshot 1`] = `
<div
  className="h-full"
/>
`;
